prism_compile.c

```diff
@@ -4684,21 +4684,21 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
                     pm_node_t *value = cast->value;
                     name = cast->name;
 
-                    if (pm_static_literal_p(value)) {
```

Comment:
I think this check was okay, could you reuse it but explicitly ignore hashes, arrays, and ranges? I think something we could do to make this more clear on the prism side is to make hashes, arrays, and ranges have different flags from static literal so that we don't have to keep ignoring them.